src/AbstractFormatTranslator.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Dhii\I18n;
+
+use ArrayAccess as Map;
+use Dhii\Data\ValueAwareInterface as Value;
+use Dhii\Util\String\StringableInterface as Stringable;
+use Dhii\I18n\Exception\TranslationExceptionInterface;
+use Dhii\I18n\Exception\I18nExceptionInterface;
+
+/**
+ * Common functionality for format string translators.
+ *
+ * @since [*next-version*]
+ */
+abstract class AbstractFormatTranslator extends AbstractStringTranslator
+{
+    /**
+     * Translates a string format.
+     *
+     * @since [*next-version*]
+     *
+     * @param string|Stringable $string  The string to translate.
+     * @param string|Value|null $context The context for the string, if any.
+     * @param string|Map|null   $params  Format parameters for interpolation.
+     *
+     * @throws TranslationExceptionInterface If problem translating.
+     * @throws I18nExceptionInterface        If a problem not directly related to translating occurs.
+     *
+     * @return string The translated string, with parameters interpolated.
+     */
+    protected function _translate($string, $context = null, $params = null)
```

Comment:
Ah, never mind. I see now that this class is extending `AbstractStringTranslator`.